include/uprotocol-cpp-ulink-zenoh/usubscription/internal/uSubscriptionClientDb.h

```diff
@@ -133,61 +105,193 @@ class USubscriptionClientDb : public UListener {
+        }
+
     private:
 
+        /**
+         * Map the created topic and callback for the notifications.
+         */
         unordered_map<std::string, notifyFunc> notifyMap_;
-
+        /**
+         * Map the publish topic and status.
+         */
         unordered_map<std::string, UCode> pubStatusMap_;
-
+        /**
+         * Map the subscribed topic and Subscription Status.
+         */
         unordered_map<std::string, SubscriptionStatus_State> subStatusMap_;
 
+
+    class uSubscriptionListener : public UListener {
+
+    public:
+
+        /**
+         * Get the instance of the listener.
+         *
+         * @return Return the instance of the listener.
+         */
+        static uSubscriptionListener& instance() noexcept {
+
+            static uSubscriptionListener listener;
+            return listener;
+        }
+
+        /**
+         * Initialize the listener.
+         *
+         * @return Retrun UCode_OK on success and ERROR on failure.
+         */
+        UCode init() {
+            if (UCode::OK != ZenohUTransport::instance().init().code()) {
+                spdlog::error("ZenohUTransport::instance().init failed");
+                return UCode::INTERNAL;
+            }
+
+            if (UCode::OK != ZenohUTransport::instance().registerListener(uSubUpdatesUri, uSubscriptionListener::instance()).code()){
+                spdlog::error("ZenohUTransport::instance().registerListener failed");
+                return UCode::INTERNAL;
+            }
+
+            return UCode::OK;
+        }
+
+        /**
+         * Terminate the listener.
+         *
+         * @return Retrun UCode_OK on success and ERROR on failure.
+         */
+        UCode term() {
+            if (UCode::OK != ZenohUTransport::instance().unregisterListener(uSubUpdatesUri, uSubscriptionListener::instance()).code()){
+                spdlog::error("ZenohUTransport::instance().unregisterListener failed");
+                return UCode::INTERNAL;
+            }
+
+            if (UCode::OK != ZenohUTransport::instance().term().code()) {
+                spdlog::error("ZenohUTransport::instance().term failed");
+                return UCode::INTERNAL;
+            }
+
+            return UCode::OK;
+        }
+
+        /**
+         * Handle the received events.
+         *
+         * @param uri - Uri for the event.
+         * @param payload - Payload of the message.
+         * @param attributes - Attributes for the message.
+         *
+         * @return Return UStatus.
+         */
+        UStatus onReceive(const UUri &uri,
+                          const UPayload &payload,
+                          const UAttributes &attributes) const override {
+            UStatus status;
+            status.set_code(UCode::OK);
+            std::string str = uprotocol::uri::LongUriSerializer::serialize(uri);
+            std::string subUri = uprotocol::uri::LongUriSerializer::serialize(uSubUpdatesUri);
+            if (str == subUri) {
```

Comment:
As you registered the OnRecive for the following URI 
("/core.usubscription/3/subscriptions#Update");

You will always receive message on this specific URI , so this check is not needed.
Instead you need to check the payload , as it is the one that actually contains the URI that you are getting the Update for